src/lib/webpack/run-webpack.js

```diff
@@ -41,9 +41,8 @@ const devBuild = async (env, onprogress) => {
 
 		compiler.plugin('done', stats => {
 			let devServer = config.devServer;
-
 			let protocol = devServer.https ? 'https' : 'http';
-			let host = process.env.HOST || devServer.host || 'localhost';
+			let host = process.env.HOST || 'localhost';
```

Comment:
I'm not so sure about this. How about seeking another opinions? You're omitting the ability to specify host via cli argument. 

In my head, this PR is as simple as replacing 0.0.0.0 with localhost if you're on Windows (or just blindly replace it because it isn't effect *nix either) here https://github.com/developit/preact-cli/blob/beb468f3bec907b7cb1f121633e0d7e05b4b8903/src/lib/webpack/run-webpack.js#L61